### ISSUE TITLE
Expose Go template file that produces the table report

### ISSUE THREAD

**Issue by turbobobbytraykov** [anchore/grype#629]
Go template files that Grype would use internally to generate reports.:

Currently if someone wants to compose a Go template using the guidelines from the README file, there is a lot of logic that needs to be reverse engineered form the Grype code if you want to get the same output as Grype's default table (columnar) summary report.:

I am composing stylized XML (for Jenkins) and HTML (for emails) template and the data I work during templating is far less valuable in its current form than the table summary report.:

**Comment by luhring** [comment]
Hi @turbobobbytraykov! For the table output (along with the other built-in formats), Grype doesn't use the template engine. Grype's code to generate table output is here: https://github.com/anchore/grype/blob/main/grype/presenter/table/presenter.go

>I am composing stylized XML (for Jenkins) and HTML (for emails) template and the data I work during templating is far less valuable in its current form than the table summary report.

Is this because you need to have certain data values shown? Or is this because you need to create a template with different formatting for XML and HTML (such as to use columns)?

**Comment by turbobobbytraykov** [comment]
Hi, @luhring !
Thank you for the prompt response!
I saw the presenter.go code and I was hoping that instead of it, we could get a yaml template that is functions like it - with "won't fix' and all other trimmings - present and in the future.
Basically, if you have some presentation logic - which may change in the future - I'd like to be able to use a template file that I can also adapt based on your template file.
Yes, I also need to create XML and HTML columnar reports - the HTML template will have color-coding, while the XML is meant to be ingested by another process.

Very interesting - while looking for better examples for this reply, I also saw another major difference in the output.
I used the template from the READMe on one of our internal Docker images so I cannot share the Docker image unfortunately. but the output count is  325 vulnerabilities (table output) vs 565 vulnerabilitie (with template file).
Here's the breakdown:
Negligible = 52 vs 152
Unknown = 6 vs 18
Low = 39 vs 91
Medium = 66 vs 86
High = 112 vs 152
Critical = 50 vs 66
After inspecting the output, I saw that the table output uses some form of distinct, because for example, CVE-2019-1010025 is mentioned only once, whereas with my template - 6 times which look exactly the same.

I am using the latest version of Grype (0.32.0).

**Comment by luhring** [comment]
>After inspecting the output, I saw that the table output uses some form of distinct

Yes, correct — for the table output, we deduplicate rows based on the fields we're showing in that output format. So if two records are distinct only via fields that aren't shown in the table, we merge them. With template outputting, you have complete control over what you show, so you can merge records if you want, by your own strategy, or not merge at all.

>I saw the presenter.go code and I was hoping that instead of it, we could get a yaml template that is functions like it - with "won't fix' and all other trimmings - present and in the future.

Are you talking about [this logic](https://github.com/anchore/grype/blob/main/grype/presenter/table/presenter.go#L50-L55)? This is a fairly simple mapping to strings, which could be encoded as logic in a Go template if you wanted to.

**Comment by kzantow** [comment]
Hiya @turbobobbytraykov , we're looking to improve templating by providing some additional functions with better examples and documentation. Any PRs are welcome!
